File: service/app/src/main/java/io/miragon/blueprint/adapter/inbound/operaton/IssueInsurancePolicyDelegate.java
package io.miragon.blueprint.adapter.inbound.operaton;

import io.miragon.blueprint.application.port.inbound.IssueInsurancePolicyUseCase;
import io.miragon.blueprint.domain.leasing.ApplicationId;
import org.operaton.bpm.engine.delegate.DelegateExecution;
import org.springframework.stereotype.Component;

@Component
public class IssueInsurancePolicyDelegate extends BaseDelegate {

    private final IssueInsurancePolicyUseCase useCase;

    public IssueInsurancePolicyDelegate(IssueInsurancePolicyUseCase useCase) {
        this.useCase = useCase;
    }

    @Override
    protected void executeTask(DelegateExecution execution) {
        useCase.issuePolicy(ApplicationId.of(execution.getProcessBusinessKey()));
    }
}

File: service/app/src/main/kotlin/io/miragon/blueprint/adapter/inbound/operaton/IssueInsurancePolicyDelegate.kt
package io.miragon.blueprint.adapter.inbound.operaton

import io.miragon.blueprint.application.port.inbound.IssueInsurancePolicyUseCase
import io.miragon.blueprint.domain.leasing.ApplicationId
import org.operaton.bpm.engine.delegate.DelegateExecution
import org.springframework.stereotype.Component

@Component
class IssueInsurancePolicyDelegate(
    private val useCase: IssueInsurancePolicyUseCase,
) : BaseDelegate() {

    override fun executeTask(execution: DelegateExecution) {
        useCase.issuePolicy(ApplicationId.of(execution.processBusinessKey))
    }
}
